File: frontend/src/types/files.ts
export interface FileNode {
  name: string;
  path: string;
  type: 'file' | 'dir';
  size: number;
  mtime: string | null;
}

export interface FileContent {
  path: string;
  content: string;
  size: number;
  mtime: string | null;
}


const VASP_FILE_TYPE_MAP: Record<string, string> = {
  INCAR: 'incar', KPOINTS: 'kpoints', POSCAR: 'poscar',
  POTCAR: 'potcar', OUTCAR: 'outcar', OSZICAR: 'oszicar',
  'vasprun.xml': 'vasprun', WAVECAR: 'binary', CHGCAR: 'binary',
  CONTCAR: 'poscar',
};

export function getVaspFileType(filename: string): string {
  return VASP_FILE_TYPE_MAP[filename] || 'text';
}

File: backend/app/schemas/files.py
"""File operation schemas."""
from pydantic import BaseModel

class FileNode(BaseModel):
    name: str
    path: str
    type: str  # "file" | "dir"
    size: int = 0
    mtime: str | None = None

class FileContent(BaseModel):
    path: str
    content: str
    size: int = 0
    mtime: str | None = None

class FileWriteRequest(BaseModel):
    path: str
    content: str

class FileMkdirRequest(BaseModel):
    path: str

class FileOperationResponse(BaseModel):
    ok: bool = True
    message: str = ""
